grails-app/services/org/pih/warehouse/product/ProductService.groovy

```diff
@@ -335,6 +336,7 @@ class ProductService {
                             eq("synonymTypeCode", SynonymTypeCode.DISPLAY_NAME)
                         }
                     }
+                    setResultTransformer(Criteria.DISTINCT_ROOT_ENTITY)
```

Comment:
I'm interested to hear more about how testing (re: pagination) went. And while it doesn't matter, I usually put this at the top of the query closure to make it clear that we're doing sneaky shit.

Reply:
I was also afraid of this, so I'm gonna test this one locally today

Reply:
Kacper showed me when the pagination doesn't work (I thought about another case so i didn't spotted this one). I am converting this pr to draft and I'm continuing working on it.

Reply:
Thanks @kchelstowski. 

@alannadolny :clap: Let us know if you need some guidance. I'll need to do some research to figure out the right approach but we can work on this together if you hit a wall.